packages/@aws-cdk/aws-lambda-golang/lib/bundling.ts

```diff
@@ -0,0 +1,161 @@
+import * as os from 'os';
+import * as path from 'path';
+import { AssetCode, Code, Runtime } from '@aws-cdk/aws-lambda';
+import * as cdk from '@aws-cdk/core';
+import { BundlingOptions } from './types';
+import { exec, findUp, getGoBuildVersion } from './util';
+
+/**
+ * Options for bundling
+ */
+export interface BundlingProps extends BundlingOptions {
+  /**
+   * Path to go.mod file
+   */
+  readonly modFilePath: string;
+
+  /**
+   * Entry path
+   */
+  readonly entry: string;
+
+  /**
+   * The runtime of the lambda function
+   */
+  readonly runtime: Runtime;
+}
+
+/**
+ * Bundling
+ */
+export class Bundling implements cdk.BundlingOptions {
+
+  public static bundle(options: BundlingProps): AssetCode {
+    return Code.fromAsset(path.dirname(options.modFilePath), {
+      assetHashType: options.assetHashType ?? cdk.AssetHashType.OUTPUT,
+      bundling: new Bundling(options),
+    });
+  }
+
+  public static clearRunsLocallyCache(): void { // for tests
+    this.runsLocally = undefined;
+  }
+
+  private static runsLocally?: boolean;
+
+  // Core bundling options
+  public readonly image: cdk.BundlingDockerImage;
+  public readonly command: string[];
+  public readonly environment?: { [key: string]: string };
+  public readonly local?: cdk.ILocalBundling;
+
+  private readonly relativeEntryPath: string;
+
+  constructor(private readonly props: BundlingProps) {
+    Bundling.runsLocally = Bundling.runsLocally
+      ?? getGoBuildVersion()
+      ?? false;
+
+    const projectRoot = path.dirname(props.modFilePath);
+    this.relativeEntryPath = `./${path.relative(projectRoot, path.resolve(props.entry))}`;
+
+    const cgoEnabled = props.cgoEnabled ? '1' : '0';
+
+    const environment = {
+      ...props.environment,
+      CGO_ENABLED: cgoEnabled,
+      GO111MODULE: 'on',
+      GOARCH: 'amd64',
+      GOOS: 'linux',
+    };
```

Comment:
```suggestion
    const environment = {
      CGO_ENABLED: cgoEnabled,
      GO111MODULE: 'on',
      GOARCH: 'amd64',
      GOOS: 'linux',
      ...props.environment,
    };
```

allow the user to override any of those values?

Reply:
agree.

Reply:
I originally decided to not allow the user to override these values because I can't think of a use case where you would want to. 

`GO111MODULE=on` - I don't think I want to allow the user to change this. I am requiring the use of Go modules and it might break the build if someone were to try to set this to `off` or `auto`.

`GOARCH=amd64` and `GOOS=linux` when building for AWS Lambda, when would I want these to be a different value (or is it even possible for them to be a different value)?

Reply:
@corymhall you know Go better than me... was just a suggestion.